pep-0698.rst

```diff
@@ -0,0 +1,439 @@
+as type annotations, which have important runtime uses such as metaprogramming
+and runtime type checking.
+
+In light of the downsides described above, we decided the benefits are
+insufficient to justify runtime behavior.
+
+
+
+Mark a base class to force explicit overrides on subclasses
+-----------------------------------------------------------
+
+We considered including a class decorator ``@require_explicit_overrides``, which
+would have provided a way for base classes to declare that all subclasses must
+use the ``@override`` decorator on method overrides. The overrides library has a
+mixin class, ``EnforceExplicitOverrides``, which provides similar behavior in
+runtime checks.
+
+We decided against this because we expect owners of large codebases will benefit
+most from ``@override``, and for these use cases having a strict mode where
+explicit ``@override`` is required (see the Backward Compatibility section)
+provides more benefits than a way to mark base classes.
+
+Moreover we believe that authors of projects who do not consider the extra type
+safety to be worth the additional boilerplate of using ``@override`` should not
+be forced to do so. Having an optional strict mode puts the decision in the
+hands of project owners, whereas the use of ``@require_explicit_overrides`` in
+libraries would force project owners to use ``@override`` even if they prefer
+not to.
+
+
+Reference Implementation
+========================
+
+Pyre: A proof of concept is implemented in Pyre:
+
+- The decorator
+  `@pyre_extensions.override <https://github.com/facebook/pyre-check/blob/f4d3f676d17b2e59c4c55d09dfa3caead8ec2e7c/pyre_extensions/__init__.py#L95/>`_
+  can mark overrides
+- Pyre can `type-check this decorator <https://github.com/facebook/pyre-check/blob/ae68c44f4e5b263ce0e175f0798272d9318589af/source/analysis/test/integration/methodTest.ml#L2515-L2638/>`_
+  as specified in this PEP
+
+
+References
+==========
+
+
+Static Override Checks in Other Languages
+-----------------------------------------
+
+Many popular programming languages support override checks. For example:
+
+- `Typescript has <https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#override-and-the---noimplicitoverride-flag/>`_ ``override``.
+- `C++ has <https://en.cppreference.com/w/cpp/language/override/>`_ ``override``.
+- `Java has <https://docs.oracle.com/javase/tutorial/java/IandI/override.html/>`_ ``@Override``.
+- `C# has <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/override/>`_ ``override``.
+- `Kotlin has <https://kotlinlang.org/docs/inheritance.html#overriding-methods/>`_ ``override``.
+- `Scala has <https://www.javatpoint.com/scala-method-overriding/>`_ ``override``.
+- `Swift has <https://docs.swift.org/swift-book/LanguageGuide/Inheritance.html#ID198/>`_ ``override``.
+- `Hack has <https://docs.hhvm.com/hack/attributes/predefined-attributes#__override/>`_ ``<<__Override>>``.
```

Comment:
Maybe sort this lexicographically, or group by the name of the deocrator, rather than seemingly arbitrary ordering?